src/main/java/unrc/dose/Proposition.java

```diff
@@ -1,18 +1,258 @@
+     */
+    public void saveSolution(final String sourceCurrent,
+            final Integer distanceObtained) {
+        this.set("source", sourceCurrent);
+        this.set("distance", distanceObtained);
+        this.set("isSubmit", 1);
+    }
+
+    /**
+     * This method find the proposition that is not a solution yet.
+     * @param userId represents the user
+     * @param challengeId represent the challenge
+     * @return null if there is no proposition or the proposition,
+     *  or the proposition found.
+     */
+    public static LazyList<Proposition> getPropositionNoSubmit(final int userId,
+            final int challengeId) {
+        return Proposition.where("user_id = ? and challenge_id = ? and "
+                + "isSubmit = ?", userId, challengeId, 0);
+    }
+
+    /**
+     * This method find the proposition that is solution
+     * for one specific challenge.
+     * @param challengerId represent the challenge.
+     * @return null if there is no solution for the challenge,
+     * else one list with all solutions.
+     */
+    public static LazyList<Proposition> allSolChall(final int challengerId) {
+        return Proposition.where("challenge_id = ? and isSubmit = ?",
+                challengerId, 1);
+    }
+
+    /**
+     * This method calculates the editing distance of a Proposition.
+     * @param p represents a Proposition
+     * @return the editing distance of both string
+     */
+    public static int getDistanceProposition(final Proposition p) {
+        String str1 = p.getSource();
+        Challenge challenge = Challenge.findById(p.get("challenge_id"));
+        String str2 = challenge.getString("source");
+
+        return computeLevenshteinDistance(str1, str2);
+    }
+
+    /**
+     * This method calculates the distance of editing between two string.
+     * @param str1 represents the original string
+     * @param str2 represents the modified string
+     * @return the editing distance of both string
+     */
+    public static int computeLevenshteinDistance(final String str1,
+            final String str2) {
+        return computeLevenshteinDistance(str1.toCharArray(),
+                str2.toCharArray());
+    }
+
+    /**
+     * Get the minimum between three characters.
+     * @param a represent a integer value
+     * @param b represent a integer value
+     * @param c represent a integer value
+     * @return represent the minimum value.
+     */
+    private static int minimum(final int a, final int b, final int c) {
+        if (a <= b && a <= c) {
+            return a;
+        }
+        if (b <= a && b <= c) {
+            return b;
+        }
+        return c;
+    }
+
+    /**
+     * This method calculates the distance of editing
+     * between two character arrangements.
+     * @param str1 represents the original arrangement
+     * @param str2 represents the modified arrangement
+     * @return the editing distance of both arrangements
+     */
```

Comment:
The name of the parameters are not consistent with the description, but the description should not mention original and modified string, since Levenshtein distance is between two strings, independently if one is the modification of the other.